extism/c-pdk
# docs/test/build: use EXTISM_EXPORTED_FUNCTION

* README.md
  * now uses `EXTISM_EXPORTED_FUNCTION` to export functions
  * add section on export details
  * Improve example code quality, for example removing stack allocated VLAs that could overflow if the input is too large

* examples
  * Switch everything but `infinite-loop` to use `EXTISM_EXPORTED_FUNCTION` as `infinite-loop.c` doesn't use the pdk.
  * Remove `.wasm` from git
 
* tests
  * fix tests to use $(WASI_SDK_PATH)/bin/clang to build as system clang doesn't work for me
  * Switch everything to use `EXTISM_EXPORTED_FUNCTION` or `EXTISM_EXPORT_AS`
* Add `.clang-format` This appears to the style already used. I had to add it locally to avoid accidentally making big reformats.
